tests/queries/0_stateless/03036_dynamic_read_subcolumns.sh

```diff
@@ -7,7 +7,7 @@ CLICKHOUSE_LOG_COMMENT=
 # shellcheck source=../shell_config.sh
 . "$CUR_DIR"/../shell_config.sh
 
-CH_CLIENT="$CLICKHOUSE_CLIENT --allow_experimental_variant_type=1 --use_variant_as_common_type=1 --allow_experimental_dynamic_type=1"
+CH_CLIENT="$CLICKHOUSE_CLIENT --allow_experimental_variant_type=1 --use_variant_as_common_type=1 --allow_experimental_dynamic_type=1 optimize_functions_to_subcolumns=0"
```

Comment:
We enabled `optimize_functions_to_subcolumns` by default recently. But `Variant` and `Dynamic` don't support `null` subcolumn and it leads to failed queries with `isNull(variant.UInt64)/isNull(dynamic.UInt64)`. I will fix it in a separate PR

Reply:
https://github.com/ClickHouse/ClickHouse/pull/66178